tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1136,14 +1155,32 @@ const reducer = createReducer(
       cardStateMap: nextCardStateMap,
     };
   }),
-  on(actions.stepSelectorToggled, (state, {affordance}) => {
+  on(actions.stepSelectorToggled, (state, {affordance, cardId}) => {
+    const nextCardStateMap = {...state.cardStateMap};
+    if (cardId) {
+      nextCardStateMap[cardId] = {
+        ...nextCardStateMap[cardId],
+        stepSelection: CardSelectionState.DISABLED,
+      };
+    } else {
+      Object.keys(nextCardStateMap).forEach((cardId) => {
```

Comment:
Curious, why use reduce for rangeSelectionToggled and forEach here? It would be nice to have the code more similar if it makes sense.